### PR TITLE
CSHARP-5087 Bump MacOS version used for tests

### PR DESCRIPTION
**Pull request by sanych-sun**

### CODE REVIEW

File: evergreen/evergreen.yml [hunk 1]
```diff
@@ -1867,40 +1873,35 @@ axes:
         display_name: "Windows 64-bit"
         variables:
           OS: "windows-64"
-          python3_binary: "C:/python/Python38/python.exe"
```
Comment (sanych-sun) [inline evergreen/evergreen.yml:1870]: I've checked our scripts and drivers-evergreen-tools and did not find any usage of the variable.

File: build.config [hunk 2]
```diff
@@ -1,3 +1,3 @@
 #!/usr/bin/env bash
 CAKE_VERSION=2.2.0
-DOTNET_VERSION=6.0.400
+DOTNET_VERSION=8.0.204
```
Comment (BorisDog) [inline build.config:3]: Finally!

File: evergreen/evergreen.yml [hunk 3]
```diff
@@ -1867,40 +1873,35 @@ axes:
...
         display_name: "Ubuntu 18.04"
         variables:
           OS: "ubuntu-1804"
-          python3_binary: "/opt/python/3.8/bin/python3"
         run_on: ubuntu1804-test
       - id: "ubuntu-2004"
         display_name: "Ubuntu 20.04"
         variables:
           OS: "ubuntu-2004"
-          python3_binary: "/opt/python/3.8/bin/python3"
```
Comment (BorisDog) [inline evergreen/evergreen.yml:1884]: Make sure that `jq` is still available after this change

**Review by BorisDog — CHANGES_REQUESTED** [review]
Small comment about jq

File: evergreen/evergreen.yml (hunk 3 above)
Comment (sanych-sun) [reply evergreen/evergreen.yml:1884]: Yep, it is. I've include the `package_push_myget` into the PR patch and it's succeeded.

**Review by BorisDog — APPROVED** [review]
LGTM

File: evergreen/evergreen.yml (hunk 1 above)
Comment (JamesKovacs) [reply evergreen/evergreen.yml:1870]: If you `git blame`, you'll find my name on the `python3_binary` lines. I had to add the `python3_binary` variable to work around `python3` detection issues on different variants. Looks like `driver-evergreen-tools` has fixed the issue and this is no longer required.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**